src/Components/Web/src/Forms/InputBase.cs

```diff
@@ -216,10 +217,37 @@ public override Task SetParametersAsync(ParameterView parameters)
                     $"{nameof(Forms.EditContext)} dynamically.");
             }
 
+            SetAdditionalAttributesIfValidationFailed();
+
             // For derived components, retain the usual lifecycle with OnInit/OnParametersSet/etc.
             return base.SetParametersAsync(ParameterView.Empty);
         }
 
+        private void OnValidateStateChanged(object sender, ValidationStateChangedEventArgs eventArgs)
+        {
+            SetAdditionalAttributesIfValidationFailed();
+
+            StateHasChanged();
+        }
+
+        private void SetAdditionalAttributesIfValidationFailed()
+        {
+            if (EditContext.GetValidationMessages(FieldIdentifier).Any())
+            {
+                // To make the `Input` components accessible by default
+                // we will automatically render the `aria-invalid` attribute when the validation fails
+                if (AdditionalAttributes == null)
```

Comment:
You can do this without requiring a breaking API change 

```C#
    Dictionary<string, object> additionalAttributes;
    if (AdditionalAttributes != null)
    {
      if (!(AdditionalAttributes is Dictionary<string, object> existing))
      {
         additionalAttributes = existing;
      }
      else
      {
        additionalAttributes = new Dictionary<string, object>(AdditionalAttributes);
      }
    }
    else
    {
        additionalAttributes = new Dictionary<string, object>();
    }

   ...

  AdditionalAttributes = additionalAttributes;
            
```

Reply:
This was my original approach, but @SteveSandersonMS suggested to make the change to avoid perf implications.

Reply:
This is what I tried originally, but it didn't work due to the Dictionary constructor accepting IDictionary<K,V> instead of IReadyOnlyDictionary<K,V>:
![image](https://user-images.githubusercontent.com/34246760/85158802-8263fb00-b211-11ea-82ff-e25cb91b52d9.png)

That's why I ended up (in my original approach) copying the values into a new dictionary, which had some perf downside. As a result, @SteveSandersonMS has suggested to go with the current approach to avoid perf hit.

Reply:
You can manually copy this in. It's really unusual to make a API breaking change without demonstrating that this actually affects perf.